src/onnx/parse_quantizelinear.cpp

```diff
@@ -37,47 +38,74 @@ struct parse_quantizelinear : op_parser<parse_quantizelinear>
     std::vector<op_desc> operators() const { return {{"QuantizeLinear"}}; }
 
     instruction_ref parse(const op_desc& opd,
-                          const onnx_parser& /*parser*/,
+                          const onnx_parser& parser,
                           const onnx_parser::node_info& info,
-                          const std::vector<instruction_ref>& args) const
+                          std::vector<instruction_ref>& args) const
     {
+        if(args.size() < 2 or args.size() > 3)
+        {
+            MIGRAPHX_THROW("QuantizeLinear: must have either 2 or 3 inputs, " +
+                           std::to_string(args.size()) + " inputs provided");
+        }
+
+        // Starting with version 19 ONNX introduced the constraint that x and y_scale types must be
+        // the same
+        if(parser.opset_version >= 19 and
```

Comment:
We have to flag it because the onnx spec states that it's a constraint for versions 19 and up. 
The common type derivation and conversion could be done for all cases, without a version check condition, but It'd be doing the extra work of common type calculation and looping over the arguments for opset versions 19+ to no avail, since we already know that the types are the same for that case.